lib/types/collection.js

```diff
@@ -0,0 +1,113 @@
+var Attribute = require('../attribute'),
+    Collection = require('../collection'),
+    CollectionAttribute;
+
+/**
+ * @class CollectionAttribute
+ * @extends Attribute
+ */
+CollectionAttribute = Attribute.inherit(/** @lends CollectionAttribute.prototype */{
+
+    /**
+     * @type {?Function}
+     * @constructs Model
+     */
+    modelType: null,
+
+    /**
+     * @type {?Function}
+     * @constructs Collection
+     */
+    collectionType: null,
+
+    default: null,
+
+    __constructor: function () {
+        this.__base.apply(this, arguments);
+
+        this._initCollection();
+    },
+
+    /**
+     * @override {Attribute}
+     */
+    isChanged: function (branch) {
+        return this.value.isChanged(branch);
+    },
+
+    /**
+     * @override {Attribute}
+     */
+    commit: function (branch) {
+        var result = this.value.commit(branch);
+        if (result) {
+            this._emitCommit(branch);
+        }
+        return result;
+    },
+
+    /**
+     * @override {Attribute}
+     */
+    revert: function (branch) {
+        return this.value.revert(branch);
+    },
+
+    /**
+     * @param {*} value
+     * @returns {Boolean}
+     */
+    isEqual: function (value) {
+        return value === null && this.value === null ? true :  this.__base(value);
+    },
+
+    toJSON: function () {
+        return this.value.toJSON();
+    },
+
+    /**
+     * @param {(Collection|Array.<(Model|Object)>)} data
+     * @returns {Collection}
+     */
+    _toAttributeValue: function (data) {
+        var CollectionClass;
+
+        if (data instanceof Collection) {
+            return data;
+        } else {
+            CollectionClass = this.collectionType ? this.collectionType : Collection;
+            if (this.modelType) {
+                CollectionClass = CollectionClass.inherit({
+                    modelType: this.modelType
+                });
+            }
+            return new CollectionClass(data);
+        }
+    },
+
+    /**
+     * @param {?Array.<(Model|Object)>} value
+     */
+    set: function (value) {
+        if (this.isEqual(value)) {
```

Comment:
At least we should implement previous method, that throws error with message, that it's not implemented/can not be implemented